librocksdb_sys/crocksdb/c.cc

```diff
@@ -699,6 +700,26 @@ struct crocksdb_file_system_inspector_t {
   std::shared_ptr<FileSystemInspector> rep;
 };
 
+struct crocksdb_post_write_callback_t : public PostWriteCallback {
+  void* state_;
+  void (*on_post_write_callback)(void*, uint64_t);
+
+  void Callback(SequenceNumber seq) override {
+    on_post_write_callback(state_, seq);
+  }
+};
+
+crocksdb_post_write_callback_t* crocksdb_post_write_callback_init(
+    void* buf, size_t buf_len, void* state,
+    on_post_write_callback_cb on_post_write_callback) {
+  assert(buf_len == sizeof(crocksdb_post_write_callback_t));
+  assert(alignof(buf) == alignof(crocksdb_post_write_callback_t));
```

Comment:
I don't think this is a valid implementation. On one hand, `alignof` should only accept a type instead of a value, on the other hand, even it accepts a value, alignof(buf) should be equal to alignof(void*), it has nothing to do with the memory buf points to. You may need to use `std::align` to check if pointer is changed.